modules/contacts/hm-contacts.php

```diff
@@ -142,13 +142,66 @@ public function reset() {
         $this->data = array();
     }
 
-    public function page($page, $size) {
+    public function page($page, $size, $data = null) {
         if ($page < 1) {
             return array();
         }
-        return array_slice($this->data, (($page - 1)*$size), $size, true);
+        if ($data === null) {
+            $data = $this->data;
+        }
+        return array_slice($data, (($page - 1)*$size), $size, true);
     }
 
+    public function group_by($column = 'group') {
+        if (!is_string($column) && !is_int($column) && !is_float($column) && !is_callable($column) ) {
+            trigger_error('group_by(): The key should be a string, an integer, or a callback', E_USER_ERROR);
+            return null;
+        }
+        $func = (!is_string($column) && is_callable($column) ? $column : null);
+        $_key = $column;
+        // Load the new array, splitting by the target key
+        $grouped = [];
+        foreach ($this->data as $value) {
+            
+            $column = null;
+            if (is_callable($func)) {
+                $column = call_user_func($func, $value);
+            } elseif (is_object($value)) {
+                $reflection = new ReflectionClass($value);
+                $property = $reflection->getProperty('data');
+                $property->setAccessible(true); // Make the private property accessible
```

Comment:
There is a reason why private properties are private and using reflection to bypass this is something nasty. If this is a contact object isn't $value->value($_key) enough to get you what you want?